tests/test_functions.py

```diff
@@ -326,3 +326,171 @@ def test_regexp_extract_all(spark):
+            "John",
+            {"age": 30, "gender": "M", "address": {"city": "New York", "state": "NY"}},
+            [
+                {"type": "home", "number": "555-1234"},
+                {"type": "work", "number": "555-5678"},
+            ],
+        ),
+        (
+            2,
+            "Jane",
+            {"age": 25, "gender": "F", "address": {"city": "San Francisco", "state": "CA"}},
+            [{"type": "home", "number": "555-4321"}],
+        ),
+    ]
+    schema = StructType(
+        [
+            StructField("id", IntegerType(), True),
+            StructField("name", StringType(), True),
+            StructField(
+                "details",
+                StructType(
+                    [
+                        StructField("age", IntegerType(), True),
+                        StructField("gender", StringType(), True),
+                        StructField(
+                            "address",
+                            StructType(
+                                [
+                                    StructField("city", StringType(), True),
+                                    StructField("state", StringType(), True),
+                                ]
+                            ),
+                            True,
+                        ),
+                    ]
+                ),
+                True,
+            ),
+            StructField(
+                "phone_numbers",
+                ArrayType(
+                    StructType(
+                        [
+                            StructField("type", StringType(), True),
+                            StructField("number", StringType(), True),
+                        ]
+                    ),
+                    True,
+                ),
+                True,
+            ),
+        ]
+    )
+    df = spark_session.createDataFrame(data, schema)
+
+    # Define expected output
+    expected_data = [
+        (1, "John", 30, "M", "New York", "NY", "home", "555-1234"),
+        (1, "John", 30, "M", "New York", "NY", "work", "555-5678"),
+        (2, "Jane", 25, "F", "San Francisco", "CA", "home", "555-4321"),
+    ]
+    expected_schema = StructType(
+        [
+            StructField("id", IntegerType(), True),
+            StructField("name", StringType(), True),
+            StructField("details_age", IntegerType(), True),
+            StructField("details_gender", StringType(), True),
+            StructField("details_address_city", StringType(), True),
+            StructField("details_address_state", StringType(), True),
+            StructField("phone_numbers_type", StringType(), True),
+            StructField("phone_numbers_number", StringType(), True),
+        ]
+    )
+    expected_df = spark_session.createDataFrame(expected_data, expected_schema)
+
+    # Apply function to input data
+    result_df = flatten_test(df)
+
+    # Check if result matches expected output
+    assert result_df.collect() == expected_df.collect()
```

Comment:
Can you please use `chispa` DataFrame equality comparisons that are more robust here?